pkg/metrics/metrics_test.go

```diff
@@ -0,0 +1,144 @@
+			"resolution":     "success",
+			"namespace":      "kanister",
+		},
+		{
+			"operation_type": "backup",
+			"resolution":     "success",
+			"namespace":      "N/A",
+		},
+		{
+			"operation_type": "backup",
+			"resolution":     "failure",
+			"namespace":      "kanister",
+		},
+		{
+			"operation_type": "backup",
+			"resolution":     "failure",
+			"namespace":      "N/A",
+		},
+		{
+			"operation_type": "restore",
+			"resolution":     "success",
+			"namespace":      "kanister",
+		},
+		{
+			"operation_type": "restore",
+			"resolution":     "success",
+			"namespace":      "N/A",
+		},
+		{
+			"operation_type": "restore",
+			"resolution":     "failure",
+			"namespace":      "kanister",
+		},
+		{
+			"operation_type": "restore",
+			"resolution":     "failure",
+			"namespace":      "N/A",
+		},
+	}
+	receivedCombinations, err := getLabelCombinations(boundedLabels)
+	isEqual := reflect.DeepEqual(receivedCombinations, expectedPrometheusLabels)
+	c.Assert(isEqual, Equals, true)
+	c.Assert(err, IsNil)
+
+	boundedLabels = make([]BoundedLabel, 0)
+	receivedCombinations, err = getLabelCombinations(boundedLabels)
+	c.Assert(receivedCombinations, IsNil)
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "invalid BoundedLabel list")
+
+	boundedLabels = make([]BoundedLabel, 1)
+	boundedLabels[0] = BoundedLabel{LabelName: "resolution", LabelValues: []string{"success", "failure"}}
+	expectedPrometheusLabels = []prometheus.Labels{
+		{
+			"resolution": "success",
+		},
+		{
+			"resolution": "failure",
+		},
+	}
+	receivedCombinations, err = getLabelCombinations(boundedLabels)
+	isEqual = reflect.DeepEqual(receivedCombinations, expectedPrometheusLabels)
+	c.Assert(isEqual, Equals, true)
+	c.Assert(err, IsNil)
+
+	boundedLabels = make([]BoundedLabel, 1)
+	boundedLabels[0] = BoundedLabel{LabelName: "resolution", LabelValues: nil}
+
+	receivedCombinations, err = getLabelCombinations(boundedLabels)
+	c.Assert(receivedCombinations, IsNil)
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "invalid BoundedLabel list")
+
+	boundedLabels = make([]BoundedLabel, 1)
+	boundedLabels[0] = BoundedLabel{LabelName: "resolution", LabelValues: []string{}}
+	c.Assert(receivedCombinations, IsNil)
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "invalid BoundedLabel list")
+}
+
+func (m *MetricsSuite) TestInitCounterVec(c *C) {
+	boundedLabels := make([]BoundedLabel, 2)
+	boundedLabels[0] = BoundedLabel{LabelName: "operation_type", LabelValues: []string{"backup", "restore"}}
+	boundedLabels[1] = BoundedLabel{LabelName: "resolution", LabelValues: []string{"success", "failure"}}
+	actionSetCounterOpts := prometheus.CounterOpts{
+		Name: "action_set_resolutions_total",
+		Help: "Total number of action set resolutions",
+	}
+	registry := prometheus.NewRegistry()
+	metrics, _ := registry.Gather()
+	c.Assert(metrics, IsNil)
+
+	actionSetCounterVec := InitCounterVec(registry, actionSetCounterOpts, boundedLabels)
+
+	metrics, _ = registry.Gather()
```

Comment:
```suggestion
	metrics, err = registry.Gather()
	c.Assert(err, IsNil)
```
Same question as above

Reply:
Thanks for catching this. We know `err` was going to be `nil` regardless, but it's better to check it as you pointed out.